server/controller/messageCenter/getTopic.ts

```diff
@@ -15,7 +18,21 @@ export const getTopic = async (
   },
   client: BaseProtocol = DB
 ): Promise<MessageTopic | undefined> => {
-  const { countryIso, assessment, cycle, key, includeMessages = true } = props
+  const { countryIso, assessment, cycle, key, includeMessages = true, user } = props
 
-  return MessageTopicRepository.getOneOrNone({ countryIso, assessment, cycle, key, includeMessages }, client)
+  const topic = await MessageTopicRepository.getOneOrNone(
+    { countryIso, assessment, cycle, key, includeMessages },
+    client
+  )
+
+  if (topic && user) {
+    const topicUser = await MessageTopicUserRepository.getOneOrNone({ assessment, cycle, topic, user })
+    if (topicUser) {
+      await MessageTopicUserRepository.update({ assessment, cycle, topic, user })
+    } else {
+      await MessageTopicUserRepository.create({ assessment, cycle, topic, user })
+    }
```

Comment:
this code is duplicated.
create a file in the same folder that export a function called updateTopicReadTime
this updateTopicReadTime will be used also in addMessage

Reply:
ok